internal/witcli/dir.go

```diff
@@ -0,0 +1,28 @@
+package witcli
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+)
+
+// FindOrCreateDir attempts to load file information for the provided path and
+// the provided path does not already exist, it attempts to create the directory
+// and then return the file information for it.
+func FindOrCreateDir(path string) (fs.FileInfo, error) {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		err = os.MkdirAll(path, os.FileMode(os.ModeDir|os.ModePerm))
```

Comment:
IIUC, `os.ModePerm` is 0x777 that gives full permissions for the owner, group, and others. Do we really want it? Perhaps we only want `0755 `

Reply:
Agree

Reply:
I'd based this off of the [original comment here](https://github.com/bytecodealliance/wasm-tools-go/issues/213#issuecomment-2436257272), sounds like we're good with `0o755` instead?

If so, I'm happy to make the change, just wanted to confirm